network/peers/subnets_test.go

```diff
@@ -120,3 +122,115 @@ func TestSubnetScore(t *testing.T) {
 		})
 	}
 }
+
+func TestUpdatePeerSubnets_Removal(t *testing.T) {
```

Comment:
This is likely not the final version of the tests, I'll be working on them on https://github.com/ssvlabs/ssv/pull/1813. I just wanted this PR to have some PoC of the fix